Add "Succeed" command modifier, like "Fail Fail" but keeps error messages (#14750)
coq/coq: doc/sphinx/proof-engine/vernacular-commands.rst

      If the given :n:`@command` succeeds, then :n:`Fail @sentence`
      fails with this error message.

.. cmd:: Succeed @sentence

>    When writing tests, it can be useful to ensure a command succeeds
>    without keeping its side effects. If :n:`@sentence` fails, then
>    :n:`Succeed @sentence` fails with the same error. If :n:`@sentence`
>    succeeds then :n:`Succeed @sentence` succeeds without changing the
>    proof state. In interactive mode, the system prints a message
>    confirming the success.

Reviewer: ```suggestion
   If :n:`@sentence` succeeds, then :n:`Succeed @sentence` succeeds without changing the
   proof state.  If :n:`@sentence` fails, then :n:`Succeed @sentence` fails showing the error
   message for :n:`@sentence.
    In interactive mode, the system prints a message confirming the success.
    This command can use useful for writing tests.
```

The success case should be described first, before the failure case.  IHMO the use case for testing should come after the description (which would improve the Fail command description as well).  And I dropped some unneeded repetition for the testing use case sentence.

Maybe include the success message with the following formatting:

```
   blah blah blah

   :n:`text of the message`
     Sentence explaining the message.
```

Since we only have constructs for warning and errors but not informational messages.
